lib/aullar/styling.moon

```diff
@@ -128,24 +134,34 @@ define_class {
 
       else -- embedded styling (sub lexing)
         if #style > 0
-          sub_base = styling[s_idx + 2]\match '[^|]*|(.+)'
+          mode_name, sub_base = styling[s_idx + 2]\match '([^|]*)|(.+)'
           sub_start_offset = base + styling_start
 
           -- determine sub styling end, so we can fill it with base
           sub_end_offset = get_sub_style_end sub_start_offset, style
           if sub_end_offset -- no sub_end_offset == empty nested styling blocks
             @set sub_start_offset, sub_end_offset - 1, sub_base, no_notify
-            @apply sub_start_offset, style, base: sub_base, no_notify: true
+            @apply sub_start_offset, style, base: sub_base, no_notify: true, :markers
             styled_up_to = sub_end_offset
+            append markers,
+              name: mode_name
+              start_offset: sub_start_offset
+              end_offset: sub_end_offset + 1
```

Comment:
@shalabhc The end offset is supposed to represent one past the end of the styling ending.